util/bitstream_test.go

```diff
@@ -27,15 +49,47 @@ var test4Bits = []testDefinition{
 	{[]byte{0x01, 0xe0}, 7, 15}, // Offset which involves flowing over to a second byte
 }
 
-func TestParseByte4(t *testing.T) {
-	b, err := ParseByte4(testdata, 46)
+func TestReadByte4(t *testing.T) {
+	bs := BitStream{b: testdata, p: 46}
+
+	b, err := bs.ReadByte4()
 	assertStringsEqual(t, "Expected 4 bits to start at bit 46, but the byte array was only 6 bytes long (needs second byte)", err.Error())
 
-	b, err = ParseByte4(testdata, 80)
+	bs = BitStream{b: testdata, p: 80}
+	b, err = bs.ReadByte4()
 	assertStringsEqual(t, "Expected 4 bits to start at bit 80, but the byte array was only 6 bytes long", err.Error())
 
 	for _, test := range test4Bits {
-		b, err = ParseByte4(test.data, test.offset)
+		bs = BitStream{b: test.data, p: test.offset}
+		b, err = bs.ReadByte4()
+		assertNilError(t, err)
+		assertBytesEqual(t, byte(test.value), b)
+	}
+}
+
+var test6Bits = []testDefinition{
+	{testdata, 21, 29},          // testdata duplicate of Offset which involves flowing over to a second byte
+	{testdata, 12, 8},           // testdata duplicate of Offset which aligns with a nibble
+	{testdata, 6, 10},           // testdata duplicate of Offset which involves flowing over to a second byte
+	{[]byte{0x10}, 0, 4},        // No offset
+	{[]byte{0x92, 0x80}, 4, 10}, // Offset which aligns with a nibble
+	{[]byte{0x99}, 1, 76},       // Offset which doesn't align with a nibble.
+	{[]byte{0x01, 0xe0}, 7, 60}, // Offset which involves flowing over to a second byte
+}
+
+func TestReadByte6(t *testing.T) {
+	bs := BitStream{b: testdata, p: 46}
+
+	b, err := bs.ReadByte6()
+	assertStringsEqual(t, "Expected 6 bits to start at bit 46, but the byte array was only 6 bytes long (needs second byte)", err.Error())
```

Comment:
Why are these cases not part of the table?

Reply:
Need a separate testing function for the errors. I had been pondering tabling them as well though.

Reply:
Might be a good idea.